src/runtime_src/core/tools/xbreplay/src/utils/logger.hpp

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+#pragma once
+
+#include <iostream>
+#include <mutex>
+#include <sstream>
+#include <stdio.h>
+
+namespace xrt::tools::xbreplay {
+enum class log_level
+{
+  XBREPLAY_DEBUG = 0,
+  XBREPLAY_INFO,
+  XBREPLAY_WARNING,
+  XBREPLAY_ERROR
```

Comment:
For scoped enum (class enums) do use only lower_case for enumerators and don't bother prefixing them. e.g
```suggestion
  debug ,
  info,
  warning,
  error
```